management/src/main/scala/akka/management/AkkaManagement.scala

```diff
@@ -86,25 +102,38 @@ final class AkkaManagement(implicit system: ExtendedActorSystem) extends Extensi
   }
 
   /**
-   * Get the routes for the HTTP management endpoint.
+   * Scala API: Get the routes for the HTTP management endpoint.
    *
    * This method can be used to embed the Akka management routes in an existing Akka HTTP server.
    */
   def routes: Try[Route] = prepareCombinedRoutes(providerSettings)
 
+  // FIXME should `routes` return `Try` of throw IllegalArgumentException?
+
+  /**
+   * Java API: Get the routes for the HTTP management endpoint.
+   *
+   * This method can be used to embed the Akka management routes in an existing Akka HTTP server.
+   * @throws IllegalArgumentException if routes configured for akka management
+   */
+  def getRoutes: akka.http.javadsl.server.Route = RouteAdapter(routes.get)
+
   /**
    * Start an Akka HTTP server to serve the HTTP management endpoint.
    */
-  // FIXME make it accept config object that would have all the `withHttps`
   def start(): Future[Uri] = {
+    // FIXME API make it accept config object that would have all the `withHttps`
```

Comment:
@chbatey What do you think about this one? I think it has major influence of the API? See `setHttpsContext` and `setHttpsContext` above?

Should it be moved to the `ManagementRouteProviderSettings` and should this method then take that as parameter. Same parameter is then need for `routes`, which is the alternative way of using this class.

Reply:
I think it is simpler to offer an overloaded API rather than two methods that need to be called in a certain order. But also don't think it is too important as it is something a user learns once and then never looks at it again